Macro Replacements.

Cameron653 opened the pull request:
-----------------------------------------**Macro Replacements**------------------------------------------------

- Replaces \red \green \blue and \black with ```<font color='COLOR'>```
- Replaces /bold with ```<B>``` and ```</B>```in a few places
- Adds a few minor coder comments in places

I did a PR recently that involved me copy and pasting old code to modify it and found out that Travis fails you if you use \red \green or \blue

As a result, I have decided I would replace as many \red \blue and \green 's as possible.

Due to the fact I have to put ```</font>``` at the end of each line I do this to, I had to do it manually and couldn't automate it. This means I had to manually go through files looking for \red \blue \green and replace it.

I wouldn't wish this torture on my worst enemy.

Not WIP anymore. Should be ready for merge.

Cameron653 commented:
The thing that shows how many more I have to go is line 267 where it says how many escapes there are.

I'm also using this to see where I put too many closing tags, as shown by lines 277 and below.

Cameron653 commented:
I replaced 655 of the 874 macros. I would replace the rest, but Github's search function isn't the best. Once this is merged I can later use the search function to find the rest of the \blue \green and \red 

Also, I apologize in advance for whoever has to look over the files before merging this. Only thing that I snuck in is a few coder comments.

SpadesNeil commented on code/game/machinery/computer/communications.dm:
```diff
@@ -53,7 +53,7 @@
 		return 1
 	if (using_map && !(src.z in using_map.contact_levels))
-		usr << "\red <b>Unable to establish a connection</b>: \black You're too far away from the station!"
+		usr << "<font color='red'> <b>Unable to establish a connection</b>: \black You're too far away from the station!</font>"
```
\black is still a thing?

SpadesNeil replied:
\black is still a thing?

Also the `</font>` tag should be moved before the `\black` macro.

Cameron653 replied:
Didn't realize that \black was an actual micro that existed, so I ended up overlooking it. Going to replace it.

SpadesNeil commented on code/game/machinery/computer/communications.dm:
```diff
@@ -275,7 +275,7 @@
 		return
 	if (using_map && !(src.z in using_map.contact_levels))
-		user << "\red <b>Unable to establish a connection</b>: \black You're too far away from the station!"
+		user << "<font color='red'> <b>Unable to establish a connection</b>: \black You're too far away from the station!</font>"
```
Same issue as my last comment.

SpadesNeil commented on code/modules/admin/DB ban/functions.dm:
```diff
@@ -82,7 +82,7 @@ datum/admins/proc/DB_ban_record(var/bantype, var/mob/banned_mob, var/duration =
 	var/DBQuery/query_insert = dbcon.NewQuery(sql)
 	query_insert.Execute()
-	usr << "\blue Ban saved to database."
+	usr << "<font color='blue'> Ban saved to database.</font>"
```
Extra space in front of this. In fact there's extra spaces in front of almost all of these.

Cameron653 replied:
Alright. When I originally did this, I looked at a few examples of where \<font color> was involved and saw that it had a space in it, like below
```
src.attack_log += text("\[[time_stamp()]\] <font color='orange'> Used deaf sting on [key_name(T)]</font>")
```
for example, but looking at a large majority of the lines that involve \<font color> it appears that they don't include a space. Going to fix it.

Cameron653 commented:
ALRIGHT. All points mentioned above has been taken care of. I removed the unneeded spaces, replaced /black in a few places where I saw it (I can replace more of them later on once this is merged via finding more with Github's search tool) and replaced /bold with the appropriate marks when I saw it.

SpadesNeil commented on code/game/machinery/computer/communications.dm:
```diff
@@ -53,7 +53,7 @@
 		return 1
 	if (using_map && !(src.z in using_map.contact_levels))
-		usr << "\red <b>Unable to establish a connection</b>: \black You're too far away from the station!"
+		usr << "<font color='red'><b>Unable to establish a connection</b></font>: <font color='black'>You're too far away from the station!</font>"
```
Scootch that `</font>:` over so it's `:</font>`

SpadesNeil commented on code/game/machinery/computer/communications.dm:
```diff
@@ -275,7 +275,7 @@
 		return
 	if (using_map && !(src.z in using_map.contact_levels))
-		user << "\red <b>Unable to establish a connection</b>: \black You're too far away from the station!"
+		user << "<font color='red'><b>Unable to establish a connection</b></font>: <font color='black'>You're too far away from the station!</font>"
```
Ditto.

SpadesNeil commented on code/game/objects/items.dm:
```diff
@@ -446,11 +446,11 @@ var/list/global/slot_flags_enumeration = list(
-	user.attack_log += "\[[time_stamp()]\]<font color='red'> Attacked [M.name] ([M.ckey]) with [src.name] (INTENT: [uppertext(user.a_intent)])</font>"
-	M.attack_log += "\[[time_stamp()]\]<font color='orange'> Attacked by [user.name] ([user.ckey]) with [src.name] (INTENT: [uppertext(user.a_intent)])</font>"
+	user.attack_log += "\[[time_stamp()]\]<font color='red'>Attacked [M.name] ([M.ckey]) with [src.name] (INTENT: [uppertext(user.a_intent)])</font>"
+	M.attack_log += "\[[time_stamp()]\]<font color='orange'>Attacked by [user.name] ([user.ckey]) with [src.name] (INTENT: [uppertext(user.a_intent)])</font>"
```
This is one spot there was SUPPOSED to be a space. :U

SpadesNeil commented on code/modules/admin/verbs/adminhelp.dm:
```diff
@@ -91,8 +91,8 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
-	var/mentor_msg = "\blue <b><font color=red>Request for Help: </font>[get_options_bar(mob, 4, 1, 1, 0)][ai_cl]:</b> [msg]"
-	msg = "\blue <b><font color=red>Request for Help:: </font>[get_options_bar(mob, 2, 1, 1)][ai_cl]:</b> [msg]"
+	var/mentor_msg = "<font color='blue'><b><font color=red>Request for Help: </font>[get_options_bar(mob, 4, 1, 1, 0)][ai_cl]:</b> [msg]</font>"
+	msg = "<font color='blue'><b><font color=red>Request for Help:: </font>[get_options_bar(mob, 2, 1, 1)][ai_cl]:</b> [msg]</font>"
```
This is formatted weirdly because the person who originally wrote this was a dingus. Please reformat it less weirdly since we might as well do that if we're editing these lines.

Here's a corrected version.

```
var/mentor_msg = "<b><font color=red>Request for Help: </font><font color='blue'>[get_options_bar(mob, 4, 1, 1, 0)][ai_cl]:</b> [msg]</font>"
msg = "<b><font color=red>Request for Help:: </font><font color='blue'>[get_options_bar(mob, 2, 1, 1)][ai_cl]:</b> [msg]</font>"
```

SpadesNeil commented on code/modules/admin/verbs/randomverbs.dm:
```diff
@@ -516,7 +516,7 @@ Traitors and the like can also be revived with the previous role mostly intact.
 			M.add_ion_law(input)
 			for(var/mob/living/silicon/ai/O in mob_list)
-				O << "\red " + input + "\red...LAWS UPDATED"
+				O << "<font color='red'> </font>" + input + "<font color='red'>...LAWS UPDATED</font>"
```
This is not correct. The goal with this macro was to make `input` red.

The correct line will look like this:

`O << "<font color='red'>" + input + " ...LAWS UPDATED</font>"`

Or alternatively, what I think looks better... although this doesn't make `input` red...

`O << input + "<font color='red'>... LAWS UPDATED!</font>"`

SpadesNeil commented on code/modules/reagents/reagent_containers/food/sandwich.dm:
```diff
@@ -96,6 +96,6 @@
-		H << "\red You lacerate your mouth on a [shard.name] in the sandwich!"
-		H.adjustBruteLoss(5) //TODO: Target head if human.
+		H << "<font color='red'>You lacerate your mouth on a [shard.name] in the sandwich!</font>"
+		H.adjustBruteLoss(5) //TODO: Target head if human. //This TODO has been here for 4 years.
```
![squid](https://user-images.githubusercontent.com/6484985/28179345-a16e913e-67cf-11e7-8053-272e6d2ecbd9.jpg)

SpadesNeil commented on code/modules/reagents/reagent_containers/food/snacks.dm:
```diff
@@ -130,11 +130,11 @@
 		return
 	else if (bitecount==1)
-		user << "\blue \The [src] was bitten by someone!"
+		user << "<font color='blue'> \The [src] was bitten by someone!</font>"
```
Extra space not removed in any part of snacks.dm for some reason.

Cameron653 replied:
I think I may have somehow overlooked that file. Whoops.

SpadesNeil commented on code/modules/surgery/bones.dm:
```diff
@@ -56,26 +56,26 @@
-		user.visible_message("[user] is beginning to set the bone in [target]'s [affected.name] in place with \the [tool]." , \
-			"You are beginning to set the bone in [target]'s [affected.name] in place with \the [tool].")
+		user.visible_message("[user] is beginning to set the bone in [target]'s [affected.name] in place with \the [tool].</font>" , \
+			"You are beginning to set the bone in [target]'s [affected.name] in place with \the [tool].</font>")
```
There's a close </font> tag, but no open <font> tag. I don't think this is supposed to be here at all.

Or maybe it is, and someone forgot to color it accordingly. I don't do surgery.

SpadesNeil commented on code/modules/surgery/face.dm:
```diff
@@ -116,14 +116,14 @@
-		user.visible_message("[user] is beginning to cauterize the incision on [target]'s face and neck with \the [tool]." , \
-		"You are beginning to cauterize the incision on [target]'s face and neck with \the [tool].")
+		user.visible_message("[user] is beginning to cauterize the incision on [target]'s face and neck with \the [tool].</font>" , \
+		"You are beginning to cauterize the incision on [target]'s face and neck with \the [tool].</font>")
```
Same as last post.

SpadesNeil commented on code/modules/surgery/generic.dm:
```diff
@@ -199,38 +199,38 @@
 			self_msg = "You start to separate the ribcage and rearrange the organs in [target]'s torso with \the [tool]."
 		if (target_zone == BP_GROIN)
-			msg = "[user] starts to pry open the incision and rearrange the organs in [target]'s lower abdomen with \the [tool]."
+			msg = "[user] starts to pry open the incision and rearrange the organs in [target]'s lower abdomen with \the [tool].</font>"
```
Ditto.

SpadesNeil commented on code/modules/surgery/generic.dm:
```diff
@@ -199,38 +199,38 @@
 		var/self_msg = "You start to pry open the incision on [target]'s [affected.name] with \the [tool]."
 		if (target_zone == BP_TORSO)
-			msg = "[user] starts to separate the ribcage and rearrange the organs in [target]'s torso with \the [tool]."
+			msg = "[user] starts to separate the ribcage and rearrange the organs in [target]'s torso with \the [tool].</font>"
```
Ditto.

SpadesNeil commented on code/modules/surgery/generic.dm:
```diff
@@ -259,16 +259,16 @@
-		user.visible_message("\blue [user] cauterizes the incision on [target]'s [affected.name] with \the [tool].", \
-		"\blue You cauterize the incision on [target]'s [affected.name] with \the [tool].")
+		user.visible_message("<font color='blue'>[user] cauterizes the incision on [target]'s [affected.name] with \the [tool].", \
+		"<font color='blue'>You cauterize the incision on [target]'s [affected.name] with \the [tool].")
```
Open tags. There's no </font> here at all.

SpadesNeil commented on code/modules/xenoarcheaology/tools/suspension_generator.dm:
```diff
@@ -234,7 +234,7 @@
+		usr << "<font color='red'>You cannot rotate [src], it has been firmly fixed to the floor.</font>"
 	else
 		set_dir(turn(dir, -90))
 
```
This took 40 minutes to review but it's done.

Cameron653 commented:
Alright. I think I got all of the things mentioned above.

SpadesNeil commented on code/modules/surgery/face.dm:
```diff
@@ -133,6 +133,6 @@
-		user.visible_message("\red [user]'s hand slips, leaving a small burn on [target]'s face with \the [tool]!", \
-		"\red Your hand slips, leaving a small burn on [target]'s face with \the [tool]!")
+		user.visible_message("<font color='red'>[user]'s hand slips, leaving a small burn on [target]'s face with \the [tool]!", \
+		"<font color='red'>Your hand slips, leaving a small burn on [target]'s face with \the [tool]!")
```
CLOSE ZE TAGS.

SpadesNeil commented on code/modules/surgery/bones.dm:
```diff
@@ -56,26 +56,26 @@
-			user.visible_message("\blue [user] sets the bone in [target]'s [affected.name]\red in the WRONG place with \the [tool].", \
-				"\blue You set the bone in [target]'s [affected.name]\red in the WRONG place with \the [tool].")
+			user.visible_message("<font color='blue'>[user] sets the bone in [target]'s [affected.name]<font color='red'> in the WRONG place with \the [tool].</font>", \
+				"<font color='blue'>You set the bone in [target]'s [affected.name]<font color='red'> in the WRONG place with \the [tool].</font>")
```
There needs to be another set of `</font>` before the `<font color='red'>`

Cameron653 commented:
Gah, one second! Thought I took care of the two mentioned above.

Anewbe commented:
Has this been tested?

Cameron653 commented:
It compiles and the server starts correctly, and all of the things that have been edited that I looked at correctly show up at red/green/blue with this way, but I haven't gone through _every_ single modification since that'd be absolutely insane.

Anewbe commented:
And suddenly merge conflicts...

SpadesNeil commented:
![angery](https://user-images.githubusercontent.com/6484985/28341529-2c2bdaa0-6be2-11e7-8d35-dba2e307f47d.gif)

Cameron653 commented:
the longer this PR is unmerged the more merge conflicts that are going to come up because it touches five thousand goddamn files

Please merge after I fix the conflicts

Cameron653 commented:
Ugh, I hate fixing conflicts with github. One second.

Cameron653 commented:
There, that should be good now. Noticed that a few extra lines were added in due to how github marks merge conflicts in the files with >>>>>>>master, but it should be ready for merge now.

Anewbe commented:
@Cameron653 This seems to have made ahelps lose their blue coloration for some people, any idea why?

SpadesNeil commented on code/modules/admin/verbs/adminhelp.dm:
```diff
@@ -91,8 +91,8 @@ var/list/adminhelp_ignored_words = list("unknown","the","a","an","of","monkey","
-	var/mentor_msg = "\blue <b><font color=red>Request for Help: </font>[get_options_bar(mob, 4, 1, 1, 0)][ai_cl]:</b> [msg]"
-	msg = "\blue <b><font color=red>Request for Help:: </font>[get_options_bar(mob, 2, 1, 1)][ai_cl]:</b> [msg]"
+	var/mentor_msg = "<b><font color=red>Request for Help: </font><font color='blue'>[get_options_bar(mob, 4, 1, 1, 0)][ai_cl]:</b> [msg]</font>"
+	msg = "<b><font color=red>Request for Help:: </font><font color='blue'>[get_options_bar(mob, 2, 1, 1)][ai_cl]:</b> [msg]</font>"
```
https://www.youtube.com/watch?v=UHoOfxcOv0E

Cameron653 replied:
...Why was this \blue and then <font color=red>
Gah, let me fix.

Cameron653 replied:
Wait what
the msg is included in the blue portion.... wha
